libs/pipingsidesheet/src/lib/utils-sidesheet/useGetChecklists.ts

```diff
@@ -0,0 +1,25 @@
+import { useHttpClient } from '@equinor/fusion-framework-react-app/http';
+import { useContextId } from '@cc-components/shared';
+import { useQuery } from '@tanstack/react-query';
+import { Checklist } from '@cc-components/pipingshared';
+
+export const useGetChecklists = (pipetestId: string) => {
+  const client = useHttpClient('cc-api');
+  const contextId = useContextId();
+
+  const { data, isLoading, error } = useQuery<Checklist[], Error>(
+    ['pipetest', pipetestId, 'checklists'],
+    async ({ signal }) => {
+      const response = await client.fetch(
+        `/api/contexts/${contextId}/pipetest/${pipetestId}/checklists`,
+        { signal }
+      );
+      if (!response.ok) {
+        throw new Error();
```

Comment:
For failed response its a good practice to do something like
```TS
throw new Error("Failed to get checklists", {cause: response});
```

Reply:
Done!